### PR TITLE
feat: integrate Protocol Contracts v2

### PR DESCRIPTION
**Pull request by lumtis**
# Description

This PR setup smart contract v2 environment.

We will use this branch as a feature branch for added functionality that breaks compatibility with previous design

### CODE REVIEW

**Comment by swift1337** [comment]
@coderabbitai summarize changes and structure them based on both features and packages

File: cmd/zetae2e/local/local.go [hunk 1]
```diff
@@ -295,14 +311,14 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
...
-			e2etests.TestUpdateBytecodeConnectorName,
-			e2etests.TestDepositEtherLiquidityCapName,
-			e2etests.TestCriticalAdminTransactionsName,
-			e2etests.TestPauseERC20CustodyName,
-			e2etests.TestMigrateERC20CustodyFundsName,
+			//e2etests.TestUpdateBytecodeConnectorName,
+			//e2etests.TestDepositEtherLiquidityCapName,
+			//e2etests.TestCriticalAdminTransactionsName,
+			//e2etests.TestPauseERC20CustodyName,
+			//e2etests.TestMigrateERC20CustodyFundsName,
```
Comment (gartnera) [inline cmd/zetae2e/local/local.go:321]: Need to revert or fix these before merge

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -92,6 +92,7 @@ jobs:
               core.setOutput('STATEFUL_DATA_TESTS', labels.includes('STATEFUL_DATA_TESTS'));
               core.setOutput('TSS_MIGRATION_TESTS', labels.includes('TSS_MIGRATION_TESTS'));
               core.setOutput('SOLANA_TESTS', labels.includes('SOLANA_TESTS'));
+              core.setOutput('V2_TESTS', labels.includes('V2_TESTS')); // for v2 tests, TODO: remove this once we fully migrate to v2 (https://github.com/zeta-chain/node/issues/2627)
```
Comment (gartnera) [inline .github/workflows/e2e.yml:95]: Please also run these on schedule

**Review by gartnera — CHANGES_REQUESTED** [review]

File: cmd/zetae2e/local/local.go (hunk 1 above)
Comment (lumtis) [reply cmd/zetae2e/local/local.go:321]: Yes correct, this was for local experimentation

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (lumtis) [reply .github/workflows/e2e.yml:100]: [c13c70b](https://github.com/zeta-chain/node/pull/2594/commits/c13c70b1feaa523ddb2e1b61be541f1dabc1f835)

**Review by gartnera — APPROVED** [review]

**Pull request merged**